sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/implementation/task/TaskSubmitter.java

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.compute.batch.implementation.task;
+
+import com.azure.compute.batch.models.BatchTaskAddCollectionResult;
+import com.azure.compute.batch.models.BatchTaskCollection;
+
+/**
+ * Interface that provides an abstraction for submitting tasks to a Batch job.
+ */
+public interface TaskSubmitter {
+
+    /**
+     * Submits a collection of tasks to a specified Batch job.
+     *
+     * @param jobId The ID of the job to which the tasks will be added.
+     * @param taskCollection The collection of tasks to be submitted.
+     * @return A BatchTaskAddCollectionResult representing the result of the task submission.
+     * @throws Exception if there is an error during task submission.
+     */
+    BatchTaskAddCollectionResult submitTasks(String jobId, BatchTaskCollection taskCollection) throws Exception;
```

Comment:
This means all submitters are really synchronous. Can we make this return a Mono<BatchTaskAddCollectionResult> in both the sync and async cases instead? It should be easy to go from Async to Sync, but not vice versa. If we treat both TaskSubmitter and TaskManager as async, we should be able to block on the Mono inside the sync client.

Reply:
ie: `Mono<BatchTaskAddCollectionResult>` instead of `BatchTaskAddCollectionResult`